### ISSUE TITLE
Creating Nodepool with Ephemeral OS Disk creates a disk of Standard HDD instead of SSD

### ISSUE THREAD

**Issue by davidallyoung** [Azure/AKS#2003]
**What happened**:

I provisioned a new nodepool on the standard_F8s_v2 sku with Ephemeral disk enabled and the VMSS was provisioned with a Standard HDD. I would expect it to be SSD.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
```
az aks nodepool add `
 -g 'rgname' `
 -n 'poolname' `
 --cluster-name 'clustername' `
 --node-vm-size standard_F8s_v2 `
 --node-count 1 `
 --max-pods 200 `
 --node-osdisk-type Ephemeral `
 --kubernetes-version '1.17.11'
```

**Anything else we need to know?**:
It may be the sku I'm using that is defaulting it to Standard HDD, but I believe this pool before I recreated it was a Premium SSD (without Ephemeral OS). The azure portal may also be reporting this incorrectly to me and it actually is a SSD. 

I noticed that it said the disk was Standard HDD on the "Disks" blade of the vmss.

![image](https://user-images.githubusercontent.com/8164812/100900479-849cd780-3488-11eb-9977-b2f8ba5dc3ae.png)

**Environment**:
- Kubernetes version (use `kubectl version`):
1.17.11
- Size of cluster (how many worker nodes are in the cluster?)
4
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.)
Various microservices that primarily interop through message brokers with customer facing apis, etc.
- Others:

**Comment by brobichaud** [comment]
I'm seeing the same thing when using ephemeral disks with Standard_D4as_v4. Clearly any local cache disk has to be an SSD, so could this be some sort of UI bug since there really isn't an Azure storage based Disk behind it?

**Comment by alexeldeib** [comment]
It’s an artifact of the underlying Azure compute implementation, if you do `az vmss show` and see this for the OS disk, you are good. Also the VMSS in the portal overview should say “Ephemeral OS: Cache Disk”.
```
{
  “diffDiskSettings”: {
    “option”: “Local”
  }
}
```

Refer https://docs.microsoft.com/en-us/azure/virtual-machines/ephemeral-os-disks#scale-set-template-deployment which shows it in the context of a template deployment as well.

**Comment by davidallyoung** [comment]
@alexeldeib Thank you for the clarification! To make sure I understand, the Standard HDD showing in disks is just a subsequent artifact and is not indicative of the actual local storage device that is enabling Ephemeral OS via the cache.

**Comment by brobichaud** [comment]
Agreed, thanks for the clarification.

While we have you engaged @alexeldeib, do you know if you'll be supporting Ephemeral OS Disks on temp disks in AKS soon as well? I noticed that was in Preview. For me this is a better solution as it offers me larger disks on smaller VM sizes.

**Comment by alexeldeib** [comment]
> Standard HDD showing in disks is just a subsequent artifact and is not indicative of the actual local storage device that is enabling Ephemeral OS via the cache.

Right. To really get in the weeds, my understanding is you always need a "clean" copy of the image to do operations like reimage. When you provision the VM, you also need a diff disk to write changes made during runtime. With ephemeral OS, the diff disk is exclusively local to the VM and will not persist data after e.g. host maintenance (hence the name). With non-ephemeral OS, the diff still writes back to Azure storage persistently. My understanding is that you see Standard_HDD for ephemeral OS because that's the storage class of the "clean" copy (admittedly not my product area, so don't quote me on that 🙂).

> do you know if you'll be supporting Ephemeral OS Disks on temp disks in AKS soon as well? I noticed that was in Preview. For me this is a better solution as it offers me larger disks on smaller VM sizes.

I wrestled with this exact issue when looking at it the first time, there's an unfortunately large matrix of various skus supporting things which we have to expose to end users. I don't think we will add cache disk before it's out of preview, but we will probably support it at some point after that. I'm mostly worried about the intersection with other features which may use the temp disk like https://github.com/Azure/AKS/issues/1753 (although admittedly, ephemeral os on cache disk would solve the same root problem that moving docker root to temp solves). A better example might be https://github.com/Azure/kubernetes-volume-drivers/issues/82, but fortunately I don't think that's very standard behavior.

tl;dr: unlikely before it's out of preview on the compute side, but probably sooner or later. it makes validation a bit trickier but I agree it's worthwhile.

**Comment by alexeldeib** [comment]
btw, keep the questions coming, but I'd assume this is fine to close otherwise?

**Comment by davidallyoung** [comment]
@alexeldeib Feel free to close this! You've really cleared up a ton for me and my team, thank you!
